### PR TITLE
refactor: migrate @masknet/base

### PR DESCRIPTION
**Pull request by lelenei**
## Description

Closes MF-0000

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

File: packages/web3-shared/base/src/index.ts [hunk 1]
```diff
@@ -1,2 +1,3 @@
 export * from './specs/index.js'
 export * from './helpers/index.js'
+export * from '@masknet/base'
```
Comment (guanbinrui) [inline packages/web3-shared/base/src/index.ts]: !

File: packages/web3-telemetry/src/hooks/useMountReport.ts [hunk 2]
```diff
@@ -1,5 +1,5 @@
 import { useMount } from 'react-use'
-import { type Unresolved, resolve } from '@masknet/shared-base'
+import { type Unresolved, resolve } from '@masknet/web3-shared-base'
```
Comment (guanbinrui) [inline packages/web3-telemetry/src/hooks/useMountReport.ts]:
```suggestion
import { type Unresolved, resolve } from '@masknet/base'
```

File: packages/web3-telemetry/src/hooks/useUnmountReport.ts [hunk 3]
```diff
@@ -1,6 +1,6 @@
 import { useUnmount } from 'react-use'
 import { TelemetryAPI } from '@masknet/web3-providers/types'
-import { type Unresolved, resolve } from '@masknet/shared-base'
+import { type Unresolved, resolve } from '@masknet/web3-shared-base'
```
Comment (guanbinrui) [inline packages/web3-telemetry/src/hooks/useUnmountReport.ts]:
```suggestion
import { type Unresolved, resolve } from '@masknet/base'
```

File: packages/web3-shared/base/src/index.ts (hunk 1 above)
Comment (guanbinrui) [inline packages/web3-shared/base/src/index.ts]:
```suggestion
```

File: packages/base/package.json [hunk 4]
```diff
@@ -13,10 +13,12 @@
     "types": "./dist/index.d.ts",
     "dependencies": {
         "@masknet/kit": "0.1.0",
+        "@masknet/public-api": "workspace:*",
```
Comment (guanbinrui) [inline packages/base/package.json]:
```suggestion
        "@masknet/public-api": "workspace:^",
```

**Comment by Jack-Works** [comment]
I need to revert this PR. You must write a changeset when you need to change this package. This package is also used for data types/functions that are intended to be published on NPM. In most cases you should move this to shared-base, that's our internal package.

**Pull request merged**